satellite/referrals/service_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package referrals_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/rpc/rpcstatus"
+	"storj.io/storj/private/testcontext"
+	"storj.io/storj/private/testplanet"
+	"storj.io/storj/private/testrand"
+	"storj.io/storj/satellite/referrals"
+)
+
+func TestServiceSuccess(t *testing.T) {
+	endpoint := &endpointHappyPath{}
+	tokenCount := 2
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1,
+		Reconfigure: testplanet.Reconfigure{
+			ReferralManagerServer: func(logger *zap.Logger) pb.ReferralManagerServer {
+				endpoint.SetTokenCount(tokenCount)
+				return endpoint
+			},
+		},
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+
+		satellite := planet.Satellites[0]
+
+		userID := testrand.UUID()
+		tokens, err := satellite.API.Referrals.Service.GetTokens(ctx, &userID)
+		require.NoError(t, err)
+		require.Len(t, tokens, tokenCount)
+
+		user := referrals.CreateUser{
+			FullName:      "test",
+			ShortName:     "test",
+			Email:         "test@mail.test",
+			Password:      "123a123",
+			ReferralToken: testrand.UUID().String(),
+		}
+		_, err = satellite.API.Referrals.Service.CreateUser(ctx, user)
+		require.NoError(t, err)
+	})
+}
+
+func TestServiceRedeemFailure(t *testing.T) {
+	endpoint := &endpointFailedPath{}
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1,
+		Reconfigure: testplanet.Reconfigure{
+			ReferralManagerServer: func(logger *zap.Logger) pb.ReferralManagerServer {
+				endpoint.SetTokenCount(2)
+				return endpoint
+			},
+		},
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+
+		satellite := planet.Satellites[0]
+
+		user := referrals.CreateUser{
+			FullName:      "test",
+			ShortName:     "test",
+			Email:         "test@mail.test",
+			Password:      "123a123",
+			ReferralToken: testrand.UUID().String(),
+		}
+		_, err := satellite.API.Referrals.Service.CreateUser(ctx, user)
```

Comment:
I don't think I understand why an error happens here. bc the referral token is random so it can't be redeemed? also is the `(endpoint *endpointFailedPath) RedeemToken` not used yet?

Reply:
The RedeemToken is the mock endpoint that's used to override the default mock server in testplanet

Reply:
it's not used in this test though right?

Reply:
It's used in this test. It overrides the default endpoint behavior to return an error